situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java

```diff
@@ -280,14 +278,12 @@ private void resetWorkState() {
      * Rotate windows for processor state
      */
     private void rotateWindows() {
-        previousWindow.clear();
-        previousWindow.close();
-        previousTraceGroupWindow.clear();
-        previousTraceGroupWindow.close();
+        previousWindow.delete();
+        previousTraceGroupWindow.delete();
         previousWindow = currentWindow;
-        currentWindow = new LmdbProcessorState(databasePath, getNewDbName(), ServiceMapStateData.class);
+        currentWindow = new LmdbProcessorState(new File(dbPath.getPath() + getNewDbName()), "spansDb", ServiceMapStateData.class);
```

Comment:
You could have just done String.join("/", part1, part2, par3)
In the future we should update the code to something like this. We can easily test the generation of each part and not have to worry about remembering which method will return with a /

Reply:
Yep, good point. I can make this change now